src/cpp/src/llm_pipeline.cpp

```diff
@@ -358,9 +362,15 @@ class StatefulLLMPipeline final : public LLMPipelineImplBase {
                 requests.push_back(sequence_group);
             }
 
-            Sampler sampler = Sampler(m_tokenizer);
+            if (m_sampler.get_seed() != config.rng_seed) {
+                m_sampler.set_seed(config.rng_seed);
+            }
+
             std::tie(result, m_selected_beam) = ov::genai::get_lm_encoded_results(m_model_runner, input_ids, concatenated_attention_mask, streamer_ptr,
-                                                                                  sampler, requests, position_ids, std::nullopt, m_selected_beam);
+                                                                                  m_sampler, requests, position_ids, std::nullopt, m_selected_beam);
+
+            for (auto& request : requests)
+                m_sampler.clear_request_info(request->get_request_id());
```

Comment:
You clear requests in both pipelines after `get_lm_encoded_results()`. Move it to `get_lm_encoded_results()`

Reply:
moved